File: src/main/java/ch/obermuhlner/astro/gradient/operation/SimpleChannelImageOperation.java
package ch.obermuhlner.astro.gradient.operation;

public class SimpleChannelImageOperation extends AbstractSimpleChannelImageOperation {

  private final ChannelOperation channelOperation;

  public SimpleChannelImageOperation(ChannelOperation channelOperation) {
    this.channelOperation = channelOperation;
  }

  @Override
  protected double channelOperation(double channel1, double channel2, int x, int y, int channelIndex) {
    return channelOperation.operation(channel1, channel2, x, y, channelIndex);
  }

  public interface ChannelOperation {
    double operation(double channel1, double channel2, int x, int y, int channelIndex);
  }
}

File: src/main/java/ch/obermuhlner/astro/gradient/operation/SimpleChannelImageOperation.kt
package ch.obermuhlner.astro.gradient.operation

class SimpleChannelImageOperation constructor(private val channelOperation: ChannelOperation) : AbstractSimpleChannelImageOperation() {
    override fun channelOperation(channel1: Double, channel2: Double, x: Int, y: Int, channelIndex: Int): Double {
        return channelOperation.operation(channel1, channel2, x, y, channelIndex)
    }

    open interface ChannelOperation {
        fun operation(channel1: Double, channel2: Double, x: Int, y: Int, channelIndex: Int): Double
    }
}